plugins/libvirt/kvm_io

```diff
@@ -59,22 +85,22 @@ def clean_vm_name(vm_name):
             pass
     return re.sub(r"[^a-zA-Z0-9_]", "_", vm_name)
 
+
 def fetch(vms):
     ''' Fetch values for a list of pids
     @param dictionary {kvm_pid: cleaned vm name}
     '''
-    res = {}
     for pid in vms:
-        f = open("/proc/%s/io" % pid, "r")
-        for line in f.readlines():
-            if "read_bytes" in line:
-                read = line.split()[1]
-                print "%s_read.value %s" % (vms[pid], read)
-            if "write_bytes" in line:
-                write = line.split()[1]
-                print "%s_write.value %s" % (vms[pid], write)
-                break
-        f.close()
+        with open("/proc/%s/io" % pid.decode(), "r") as f:
+            for line in f.readlines():
+                if "read_bytes" in line:
+                    read = line.split()[1]
+                    print("%s_read.value %s" % (vms[pid], read))
+                if "write_bytes" in line:
+                    write = line.split()[1]
+                    print("%s_write.value %s" % (vms[pid], write))
+                    break
```

Comment:
Just for my curiosity: why should the processing stop after `write_bytes`?
Does this imply some specific sorting of the output? Maybe the original code included this erroneously?

Reply:
it looks like it is needed.
the output in my case without the break would be not valid:
./kvm_io
KVM_NGFW_AK_read.value 7456280576
KVM_NGFW_AK_write.value 2008285184
KVM_NGFW_AK_write.value 0

whereas with the break it looks as wanted:
./kvm_io
KVM_NGFW_AK_read.value 7456280576
KVM_NGFW_AK_write.value 2008289280

Reply:
OK.
In order to avoid the same question to be researched again in the future, it would be perfect to add a comment, explaining this (based on the nature of the input data stream).
But anyway: since you planned to just do the python3 migration, it is fine to leave this old stain here ...